R/utility.R

```diff
@@ -725,4 +724,117 @@ viewSplines <- function(knots, degree, theta) {
   }
 
   return(p)
-}
\ No newline at end of file
+}
+
+#' Get survival curve parameters
+#'
+#' @param points A list of two-element vectors specifying the desired time and 
+#' probability pairs that define the desired survival curve
+#' @return A vector of parameters that define the survival curve optimized for
+#' the target points. The first element of the vector represents the "f"
+#' parameter and the second element represents the "shape" parameter.
+#' @examples
+#' points <- list(c(60, 0.90), c(100, .75), c(200, .25), c(250, .10))
+#' survGetParams(points)
+#' @export
+#' @concept utility
+survGetParams <- function(points) {
+  
+  assertNotMissing(points = missing(points))
+  assertClass(points = points, class = "list")
+  
+  time <- sapply(points, function(x) x[1])
+  p <- sapply(points, function(x) x[2])
+  
+  assertPositive(time)
+  assertAscending(time)
+  
+  assertProbability(p)  
+  assertDescending(p)
+  
+  loss_surv <- function(params, points) {
+    
+    loss <- function(a, params) {
+      ( params[2]*(log(-log(a[2])) - params[1]) - log(a[1]) ) ^ 2
+    }
+    
+    sum(sapply(points, function(a) loss(a, params)))
+    
+  }
+  
+  r <- stats::optim(
```

Comment:
It is just a style thing but one letter variable names should be avoided in most cases for readability, (I know statisticians love them :grin: ). As this is only used in two pretty unambiguous places it is not critical but worth mentioning.

Reply:
Changed to `optim_results`.